File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str = "";
        for (auto now : s) {
            if (!isalpha(now) && !('0' <= now && now <= '9')) {
                continue;
            }
            char c = now;
            if (isalpha(c)) c = tolower(c);
            str += c;
        }
        int n = str.size();
        for (int i = 0; i < n / 2; i++) {
            if (str[i] != str[n - i - 1]) {
                return false;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.go
func isPalindrome(s string) bool {
    l, r := 0, len(s)-1
    for l < r {
        for l < r && !("a" <= strings.ToLower(string(s[l])) && strings.ToLower(string(s[l])) <= "z" ||
                       '0' <= s[l] && s[l] <= '9') {
            l++
        }
        for l < r && !("a" <= strings.ToLower(string(s[r])) && strings.ToLower(string(s[r])) <= "z" ||
                       '0' <= s[r] && s[r] <= '9') {
            r--
        }
        if l >= r {
            break
        }
        if strings.ToLower(string(s[l])) != strings.ToLower(string(s[r])) {
            return false
        }
        l++
        r--
        
    }
    return true
}